test/steps/kogitoapp.go

```diff
@@ -173,3 +134,49 @@ func (data *Data) scaleKogitoApplicationToPodsWithinMinutes(name string, nbPods,
 func (data *Data) kogitoApplicationLogContainsTextWithinMinutes(dcName, logText string, timeoutInMin int) error {
 	return framework.WaitForAllPodsToContainTextInLog(data.Namespace, dcName, logText, timeoutInMin)
 }
+
+// Misc methods
+
+// getKogitoAppExampleStub Get basic KogitoApp stub with GIT properties initialized to common Kogito examples
+func getKogitoAppExamplesStub(namespace, contextDir string, native, persistence, events bool) *v1alpha1.KogitoApp {
+	kogitoApp := framework.GetKogitoAppStub(namespace, filepath.Base(contextDir))
+
+	kogitoApp.Spec.Build.GitSource.URI = config.GetExamplesRepositoryURI()
+	kogitoApp.Spec.Build.GitSource.ContextDir = contextDir
+
+	if ref := config.GetExamplesRepositoryRef(); len(ref) > 0 {
+		kogitoApp.Spec.Build.GitSource.Reference = ref
+	}
+
+	// Add namespace for service discovery
+	// Can be removed once https://issues.redhat.com/browse/KOGITO-675 is done
+	kogitoApp.Spec.KogitoServiceSpec.AddEnvironmentVariable("NAMESPACE", namespace)
```

Comment:
don't we want to put that into the `framework.GetKogitoAppStub` ?
I mean it impacts all kogitoapp as soon as they use the service discovery... but we don't need to make the distinction